src/stk/serialization/json/serializers/constructed_molecule/constructed_molecule.py

```diff
@@ -107,9 +107,17 @@ def get_keys(building_block):
         building_block_indices[None] = None
 
         def atom_info_to_json(atom_info):
+            if atom_info.get_building_block() is None:
+                return (
+                    None,
+                    None,
+                    None,
+                )
+
             return (
                 building_block_indices[atom_info.get_building_block()],
                 atom_info.get_building_block_id(),
+                atom_info.get_building_block_atom().get_id(),
```

Comment:
I assume that this ordering does not matter compared to its ordering in the arguments list?

Reply:
Correct, and it means that backwards compatibility of the serialized representation is maintained.

Reply:
Perfect. Good coding! ahha